src/main/java/org/broadinstitute/hellbender/tools/SplitReads.java

```diff
@@ -132,8 +165,13 @@ private void addKey(final List<List<?>> listKeys, final int listIndex,
      * @return The generated key that may then be used to find the appropriate SAMFileWriter.
      */
     private String getKey(final List<ReaderSplitter<?>> splitters, final GATKRead record) {
+        // if a read is missing the value for the target split, return the value "unknown" which will
+        // result in a new output stream being created on demand to hold uncategorized reads
         return splitters.stream()
-                .map(s -> s.getSplitBy(record, getHeaderForReads()).toString())
+                .map(s -> {
+                    final Object key = s.getSplitBy(record, getHeaderForReads());
+                    return key == null ? UNKNOWN_OUT_PREFIX : key.toString();
+                })
                 .reduce("", (acc, item) -> acc + "." + item);
```

Comment:
It's not your code, but I think this reduction would be more idiomatic as a `collection(Collectors.join("."))`